src/utils.cpp

```diff
@@ -3,19 +3,28 @@
 #include <vuh/arr/arrayUtils.h>
 
 #include <fstream>
+#include <iterator>
 
 namespace vuh {
+	struct UInt32 {
```

Comment:
Is this to prevent implicit casts? Anyway please add a brief comment here.

Reply:
This is to use the `istream_iterator<UInt32>` to fill the vector.  I don't know of another way to do it that adjusts the size of the `vector<uint32_t>` automatically.  I would simply `fin.read(vec.data(), filesize)` but I don't know the filesize.  To my understanding (I didn't try it) `istream_iterator<uint32_t>` expects ASCII string numbers that can be converted to `uint32_t` so I defined the type `UInt32` that reads binary numbers, instead.  Does a comment like

`// read binary uint32_t with istream_iterator<UInt32> from fstream`

explains this sufficiently?  How do I change the pull request accordantly?

Reply:
ah, right, I see, sorry I was a bit sloppy reading your CL. No need to change anything. BTW I was fixing this in the version 2 branch some time ago, you may take a look at https://github.com/Glavnokoman/vuh/blob/vuh2/src/kernel.cpp#L164. I guess I still like my version better, but I may be biased :).

Reply:
Oh, I should have a look an vuh2.  If vuh2 is C++17 you can get the filesize with `std::filesystem::file_size`.  I'm a little bit sceptical regarding the `tellg` method as it seems one has to really know what one is doing (see e.g. <https://stackoverflow.com/questions/22984956/tellg-function-give-wrong-size-of-file>).  But I would have done exactly the same using `file_size` instead of `tellg` if `filesystem` would have been available in vuh(1).

Reply:
`tellg` should be fine on files opened in binary mode. The zoo of `std::filesystem` implementations (header/linker options per compiler) creates quite a barrier to pull it into the project, so I chose to stay away from it for such a minor use.